app/src/geckoNightly/java/org/mozilla/fenix/engine/GeckoProvider.kt

```diff
@@ -18,23 +18,22 @@ import org.mozilla.geckoview.ContentBlocking
 import org.mozilla.geckoview.ContentBlocking.SafeBrowsingProvider
 
 object GeckoProvider {
+    @Volatile
     private var runtime: GeckoRuntime? = null
     const val CN_UPDATE_URL =
         "https://sb.firefox.com.cn/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2"
     const val CN_GET_HASH_URL =
         "https://sb.firefox.com.cn/gethash?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2"
 
-    @Synchronized
     fun getOrCreateRuntime(
```

Comment:
Since this method is invoked only once during startup there should be no noticeable performance difference between `Synchronized` and the `Volatile` approach here.

There should be no relevant lock contention to cause an impact. It's `Synchronized` because it could be the Gecko fetch client or the engine itself that needs it first. The fetch client could be initialized on a different thread.

In general, we don't want to optimize code before we've demonstrated a performance impact. Did you measure this or notice a difference?